incubator/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/stream/MqttKafkaPublishFactory.java

```diff
@@ -357,16 +359,16 @@ private void onMqttData(
 
             if (mqttPublishDataEx.responseTopic().asString() != null)
             {
-                final String16FW replyTopic = mqttPublishDataEx.responseTopic();
+                final String16FW responseTopic = mqttPublishDataEx.responseTopic();
                 addHeader(helper.kafkaReplyToHeaderName, kafkaMessagesTopic);
-                addHeader(helper.kafkaReplyKeyHeaderName, replyTopic);
+                addHeader(helper.kafkaReplyKeyHeaderName, responseTopic);
 
-                String[] replyTopicHeaders = replyTopic.asString().split("/");
-                for (int i = 0; i < replyTopicHeaders.length; i++)
+                AtomicInteger offset = new AtomicInteger();
```

Comment:
Use can use a `MutableInteger` instead (single threaded), and also hoist the instance to factory state field.